GeometryCore/src/Polygon.cpp

```diff
@@ -252,4 +253,64 @@ namespace geom_utils
         return poly;
     }
 
+    //return inflate poligon
+    Polygon Polygon::inflate(const float value) const 
+    {
+        Polygon poly;
+        bool polyIs = orientation();
+
+        for (int i = 0; i < points.size(); i++) 
+        {
+        
+            // get this point (pt1), the point before it (pt0) and the point that follows it (pt2)
+            FPoint2D point0 = points[(i > 0) ? i - 1 : points.size() - 1];
+            FPoint2D point1 = points[i];
+            FPoint2D point2 = points[(i + 1) % points.size()];
+
+            
+            // find the line vectors of the lines goingn to the current point
+            FPoint2D v1(point1 - point0);
+            FPoint2D v2(point2 - point1);
+            if (cross(v1, v2) != 0) {
```

Comment:
Do not compare float point numbers with strict comparison operators ==, !=. This is because of how float point numbers are represented internally. Use std::fabs() for comparison: `std::fabs(cross(v1, v2) < some_tolerance`. In our GeometryCore you can use, for example, getUnitTolerance() for tolerance value. Further explanation goes out scope of our task, but more info about float point comparison in C++ you can find in the internet.